File: app/src/main/java/com/absinthe/kage/media/Playback.java
package com.absinthe.kage.media;

public interface Playback {

    interface Callback {
        void onCompletion();

        void onError(String str);

        void onMediaMetadataChanged(LocalMedia tCastLocalMedia);

        void onPlaybackStateChanged(int i);
    }

    int getBufferPosition();

    int getCurrentPosition();

    int getDuration();

    int getState();

    void pause();

    void play();

    void playMedia(LocalMedia tCastLocalMedia);

    void seekTo(int position);

    void setCallback(Callback callback);

    void stop(boolean isStop);
}

File: app/src/main/kotlin/com/absinthe/kage/media/Playback.kt
package com.absinthe.kage.media

const val TYPE_LOCAL = 1
const val TYPE_REMOTE = 2

interface Playback {
    interface Callback {
        fun onCompletion()
        fun onError(error: String)
        fun onMediaMetadataChanged(localMedia: LocalMedia)
        fun onPlaybackStateChanged(state: Int)
    }

    val bufferPosition: Int
    val currentPosition: Int
    val duration: Int
    val state: Int

    fun pause()
    fun play()
    fun playMedia(localMedia: LocalMedia)
    fun seekTo(position: Int)
    fun setCallback(callback: Callback)
    fun stop(isStop: Boolean)
}